src/test/java/org/apache/datasketches/theta/CornerCaseThetaSetOperationsTest.java

```diff
@@ -0,0 +1,558 @@
+    EXACT,        //{ 1.0, >0, F} Bin: 111  Oct: 07, specify only value
+    ESTIMATION,   //{<1.0, >0, F} Bin: 010  Oct: 02, specify only value
+    NEW_DEGEN,    //{<1.0,  0, T} Bin: 001  Oct: 01, specify only p
+    RESULT_DEGEN  //{<1.0,  0, F} Bin: 000  Oct: 0, specify p, value
+  }
+
+  //NOTE: 0 values in getSketch are not used.
+
+  private static void checks(
+      UpdateSketch thetaA,
+      UpdateSketch thetaB,
+      double resultInterTheta,
+      int resultInterCount,
+      boolean resultInterEmpty,
+      double resultAnotbTheta,
+      int resultAnotbCount,
+      boolean resultAnotbEmpty) {
+    CompactSketch csk;
+
+    //Intersection
+    Intersection inter = SetOperation.builder().buildIntersection();
+
+    csk = inter.intersect(thetaA, thetaB);
+    checkResult("Intersect Stateless Theta, Theta", csk, resultInterTheta, resultInterCount, resultInterEmpty);
+    csk = inter.intersect(thetaA.compact(), thetaB.compact());
+    checkResult("Intersect Stateless Theta, Theta", csk, resultInterTheta, resultInterCount, resultInterEmpty);
+
+    //AnotB
+    AnotB anotb = SetOperation.builder().buildANotB();
+
+    csk = anotb.aNotB(thetaA, thetaB);
+    checkResult("AnotB Stateless Theta, Theta", csk, resultAnotbTheta, resultAnotbCount, resultAnotbEmpty);
+    csk = anotb.aNotB(thetaA.compact(), thetaB.compact());
+    checkResult("AnotB Stateless Theta, Theta", csk, resultAnotbTheta, resultAnotbCount, resultAnotbEmpty);
+
+    anotb.setA(thetaA);
+    anotb.notB(thetaB);
+    csk = anotb.getResult(true);
+    checkResult("AnotB Stateful Theta, Theta", csk, resultAnotbTheta, resultAnotbCount, resultAnotbEmpty);
+
+    anotb.setA(thetaA.compact());
+    anotb.notB(thetaB.compact());
+    csk = anotb.getResult(true);
+    checkResult("AnotB Stateful Theta, Theta", csk, resultAnotbTheta, resultAnotbCount, resultAnotbEmpty);
+  }
+
+
+  @Test
+  public void newNew() {
+    UpdateSketch thetaA = getSketch(SkType.NEW,    0, 0);
+    UpdateSketch thetaB = getSketch(SkType.NEW,    0, 0);
+    final double resultInterTheta = 1.0;
+    final int resultInterCount = 0;
+    final boolean resultInterEmpty = true;
+    final double resultAnotbTheta = 1.0;
+    final int resultAnotbCount = 0;
+    final boolean resultAnotbEmpty = true;
+
+    checks(thetaA, thetaB, resultInterTheta, resultInterCount, resultInterEmpty,
+        resultAnotbTheta, resultAnotbCount, resultAnotbEmpty);
+  }
+
+  @Test
+  public void newExact() {
+    UpdateSketch thetaA = getSketch(SkType.NEW,    0, 0);
+    UpdateSketch thetaB = getSketch(SkType.EXACT,  0, GT_MIDP_V);
+    final double resultInterTheta = 1.0;
+    final int resultInterCount = 0;
+    final boolean resultInterEmpty = true;
+    final double resultAnotbTheta = 1.0;
+    final int resultAnotbCount = 0;
+    final boolean resultAnotbEmpty = true;
+
+    checks(thetaA, thetaB, resultInterTheta, resultInterCount, resultInterEmpty,
+        resultAnotbTheta, resultAnotbCount, resultAnotbEmpty);
+  }
```

Comment:
What effect does a sampling probability of zero have on whether a sketch stores a value in its buffer?  This question is in reference to line 118 where an exact sketch is created with a value greater than the MIDP_V.

Reply:
If you look at line 53 (corrected) it reads:
`EXACT,        //{ 1.0, >0, F} Bin: 110  Oct: 06, specify only value`

And line 59:
`//NOTE: 0 values in getSketch are not used.`
Examine lines 512-515, p is not used.

So for line 118:
`UpdateSketch thetaB = getSketch(SkType.EXACT,  0, GT_MIDP_V);`

I could have used any integer value for this test, I just happened to choose GT_MIDP_V (3).  The 0 is not used.

Reply:
Makes sense, thanks.